src/lib.rs

```diff
@@ -207,6 +225,19 @@ impl Context {
     }
 }
 
+/// A rectangular region of the buffer coordinate space.
+#[derive(Clone, Copy, Debug)]
+pub struct Rect {
+    /// x coordinate of top left corner
+    pub x: i32,
+    /// y coordinate of top left corner
+    pub y: i32,
+    /// width
+    pub width: i32,
+    /// height
+    pub height: i32,
```

Comment:
```suggestion
    /// x coordinate of top left corner
    pub x: u32,
    /// y coordinate of top left corner
    pub y: u32,
    /// width
    pub width: NonZeroU32,
    /// height
    pub height: NonZeroU32,
```
Wouldn't that make more sense?
I assume you decided on `i32` because all APIs take a `i32`, but Web doesn't. In any case, I think this would be better, I'm not actually sure what a negative `x`, `y`, `width` or `height` would do.

Reply:
Windows and Wayland take `i32` here. But I expect that's just C programmers defaulting to `int` even for things that should never be negative. Presumably none of these should have negative values, and width and height should be non-zero.

Either way it requires conversion depending on the backend (like the current `NonZeroU32` sizes). I guess it can return an error for values that are out of range...